api/handler.go

```diff
@@ -406,6 +386,80 @@ func (h *apiHandler) queryTotalUsage() http.HandlerFunc {
 	}
 }
 
+func (h *apiHandler) queryRealtimeViewership() http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		querySpec, httpErrorCode, errs := h.resolveRealtimeQuerySpec(r)
+		if len(errs) > 0 {
+			respondError(rw, httpErrorCode, errs...)
+			return
+		}
+		metrics, err := h.views.QueryRealtimeEvents(r.Context(), querySpec)
+		if err != nil {
+			respondError(rw, http.StatusInternalServerError, err)
+			return
+		}
+		respondJson(rw, http.StatusOK, metrics)
+	}
+}
+
+func (h *apiHandler) resolveQuerySpec(r *http.Request) (views.QuerySpec, int, []error) {
```

Comment:
**nit:** Mind specifying that these are `viewership` query specs? We have the billing code here as well which copied most of the abstractions, so it can be ambiguous

Reply:
Update in 1fe10a08538390a1a9e6cd0d8c3d12f3ec562414